test/thread_pool.cpp

```diff
@@ -0,0 +1,64 @@
+//---------------------------------------------------------------------------//
+// 
+// Copyright (c) 2024 Martun Karapetyan <martun@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#define BOOST_TEST_MODULE thread_pool_test
+
+#include <vector>
+#include <cstdint>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/data/monomorphic.hpp>
+
+#include <nil/actor/core/thread_pool.hpp>
+
+using namespace nil::crypto3::algebra;
+using namespace nil::crypto3::math;
+
+BOOST_AUTO_TEST_SUITE(thread_pool_test_suite)
+
+BOOST_AUTO_TEST_CASE(vector_multiplication_test) {
+    size_t size = 131072;
+
+    std::vector<int> v(size);
+
+    for (int i = 0; i < size; ++i)
```

Comment:
size: size_t
i: int, 
let's make them the same type.

Reply:
Done.